### PR TITLE
Added note regarding GHES support

### PR DESCRIPTION
**Pull request by kotewar**
### What are you trying to accomplish?
Added a note about GHES support for the extension
[Rendered page](https://github.com/actions/gh-actions-cache/blob/e4c5811b8d1fe5ff8ee5f2b967ca1ffa124d3ac6/README.md)

### What approach did you choose and why?

Just a README update for the users

### Anything you want to highlight for special attention from reviewers?

No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ It also allows deleting a corrupt, incomplete or dangling cache. A cache can be
 
 This extension builds on top of [cache management](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#managing-caches) capabilities exposed by GitHub [APIs](https://docs.github.com/en/rest/actions/cache).
 
+**Note:** This extension currently supports github.com repositories only and will start supporting GHES (GitHub Enterprise Server) by the next ship version i.e. 3.7.
```
Comment (aparna-ravindra) [inline README.md:11]:
```suggestion
**Note:** This extension currently supports github.com repositories only and will start supporting GHES (GitHub Enterprise Server) starting GHES 3.7.
```

[the remark this answers is not in the document]
Comment (kotewar) [reply README.md]: Removed the sentence that states GHES and version and wrote `GitHub Enterprise Server is not supported currently.`

Comment (kotewar) [reply README.md]: Removed this as we don't want to state the version as of now.

File: README.md [hunk 2]
```diff
@@ -97,4 +99,4 @@ EXAMPLES:
 
 
 ## Contributing
-If anything feels off, or if you feel that some functionality is missing, please check out the [contributing page](CONTRIBUTING.md). There you will find instructions for sharing your feedback, building the tool locally, and submitting pull requests to the project.
\ No newline at end of file
+If anything feels off, or if you feel that some functionality is missing, please check out the [contributing page](CONTRIBUTING.md). There you will find instructions for sharing your feedback, building the tool locally, and submitting pull requests to the project.
```
Comment (t-dedah) [inline README.md:102]: There is no change here right?

Comment (kotewar) [reply README.md:102]: No change. Could be a minor formatting by the editor.

**Review by aparna-ravindra — APPROVED** [review]

**Pull request merged**